lib/purescript-cst/src/Control/Monad/Supply/Class.hs

```diff
@@ -29,5 +33,6 @@ instance Monad m => MonadSupply (SupplyT m) where
 instance MonadSupply m => MonadSupply (StateT s m)
 instance (Monoid w, MonadSupply m) => MonadSupply (WriterT w m)
 
+-- | Convenience function for returning "$x" where `x` is the next unique integer
```

Comment:
IMO this comment doesn't need to describe the implementation of `freshName`; instead it should contain a description of the sorts of situations in which `freshName` is intended to be used, as opposed to the similar `freshIdent'` in `Language.PureScript.Names`. (The facts on the ground are that `freshName` is used in the CoreImp optimizer and the JS generator; it's also used in one place in `L.PS.Linter.Exhaustive` but I'm not sure that instance shouldn't actually be `freshIdent'`—I've just never had a reason to try rewriting it.)

Reply:
Mind pushing a commit that makes that update?

Reply:
Can do, though I was hoping another maintainer might chime in to fill in the blanks I'm missing.

Reply:
Then, perhaps we remove that for now, too, and fill it in at a later time?

Reply:
I gave it a tweak; I'm like 90% sure this is correct, and I don't want to let the perfect delay the good.